### PR TITLE
Cleanup of docstrings of scales

### PR DESCRIPTION
**Pull request by timhoffm**

### CODE REVIEW

File: lib/matplotlib/scale.py [hunk 1]
```diff
@@ -100,8 +107,8 @@ def set_default_locators_and_formatters(self, axis):
 
     def get_transform(self):
         """
-        The transform for linear scaling is just the
-        :class:`~matplotlib.transforms.IdentityTransform`.
+        Return the transform for linear scaling, which is just the
```
Comment (anntzer) [inline lib/matplotlib/scale.py]: just inherit all these docstrings? not sure per-scale documentation of get_transform is really useful.

File: lib/matplotlib/scale.py [hunk 2]
```diff
@@ -172,16 +179,12 @@ def forward(values: array-like) -> array-like
         self._transform = transform
 
     def get_transform(self):
-        """
-        The transform for arbitrary scaling
-        """
+        """Return the `.FuncTransform` associated with this scale."""
+        # docstring inherited
```
Comment (anntzer) [inline lib/matplotlib/scale.py]: well, right now it's not inherited :) but per the above I would suggest to inherit it

File: lib/matplotlib/scale.py [hunk 3]
```diff
@@ -669,19 +658,22 @@ class LogitScale(ScaleBase):
 
     def __init__(self, axis, nonpos='mask'):
         """
-        *nonpos*: {'mask', 'clip'}
-          values beyond ]0, 1[ can be masked as invalid, or clipped to a number
-          very close to 0 or 1
+        Parameters
+        ----------
+        axis : `matplotlib.axis.Axis`
+            Currently used.
```
Comment (anntzer) [inline lib/matplotlib/scale.py]: *un*used

File: lib/matplotlib/scale.py [hunk 4]
```diff
@@ -719,22 +712,25 @@ def scale_factory(scale, axis, **kwargs):
...
+    axis : `matplotlib.axis.Axis`
     """
     scale = scale.lower()
     cbook._check_in_list(_scale_mapping, scale=scale)
     return _scale_mapping[scale](axis, **kwargs)
 
 if scale_factory.__doc__:
     scale_factory.__doc__ = scale_factory.__doc__ % {
-        "names": ", ".join(get_scale_names())}
+        "names": ", ".join(f"'{s}'" for s in get_scale_names())}
```
Comment (anntzer) [inline lib/matplotlib/scale.py]: e.g. `map(repr, get_scale_names())`

File: lib/matplotlib/scale.py (hunk 1 above)
Comment (timhoffm) [reply lib/matplotlib/scale.py]: For now, I'd leave it. There's a bit of value to have the link to the respective transform in the subclasses. Alternatively, one could document these relations for the builtin scales in a common place. and eliminate most of the `get_transform`s. Almost all are just returning `self._transform`. But that's for another time. You are welcome to make a PR (*after* this is merged :wink:).

File: lib/matplotlib/scale.py [hunk 5]
```diff
@@ -354,18 +356,17 @@ class LogScale(ScaleBase):
...
            The base of the logarithm
-
-        *nonposx*/*nonposy*: {'mask', 'clip'}
-          non-positive values in *x* or *y* can be masked as
-          invalid, or clipped to a very small positive number
-
-        *subsx*/*subsy*:
+        nonposx, nonposy : {'mask', 'clip'}
+            Determines the behavior for non-positive values. They can either
+            be masked as invalid, or clipped to a number very small positive
```
Comment (QuLogic) [inline lib/matplotlib/scale.py]:
```suggestion
            be masked as invalid, or clipped to a very small positive
```

File: lib/matplotlib/scale.py [hunk 6]
```diff
@@ -669,19 +657,22 @@ class LogitScale(ScaleBase):
...
         """
-        *nonpos*: {'mask', 'clip'}
-          values beyond ]0, 1[ can be masked as invalid, or clipped to a number
-          very close to 0 or 1
+        Parameters
+        ----------
+        axis : `matplotlib.axis.Axis`
+            Currently unused.
+        nonpos : {'mask', 'clip'}
+            Determines the behavior for values beyond ]0, 1[. They can either
```
Comment (QuLogic) [inline lib/matplotlib/scale.py]: I think another PR suggested that we should explicitly say "open interval" or "closed interval" for these sorts of things.

File: lib/matplotlib/transforms.py [hunk 7]
```diff
@@ -1403,11 +1403,17 @@ def transform_affine(self, values):
         affine transformations, this is equivalent to
         ``transform(values)``.
 
-        Accepts a numpy array of shape (N x :attr:`input_dims`) and
-        returns a numpy array of shape (N x :attr:`output_dims`).
+        Parameters
+        ----------
+        values : array
+            The input values as numpy array of length :attr:`input_dims` or
```
Comment (QuLogic) [inline lib/matplotlib/transforms.py]:
```suggestion
            The input values as NumPy array of length :attr:`input_dims` or
```

File: lib/matplotlib/transforms.py [hunk 8]
```diff
@@ -1403,11 +1403,17 @@ def transform_affine(self, values):
...
+            shape (N x :attr:`input_dims`).
 
-        Alternatively, accepts a numpy array of length :attr:`input_dims`
-        and returns a numpy array of length :attr:`output_dims`.
+        Returns
+        -------
+        values : array
+            The output values as numpy array of length :attr:`input_dims` or
```
Comment (QuLogic) [inline lib/matplotlib/transforms.py]:
```suggestion
            The output values as NumPy array of length :attr:`input_dims` or
```

File: lib/matplotlib/transforms.py [hunk 9]
```diff
@@ -1422,11 +1428,17 @@ def transform_non_affine(self, values):
         ``transform(values)``.  In affine transformations, this is
         always a no-op.
 
-        Accepts a numpy array of shape (N x :attr:`input_dims`) and
-        returns a numpy array of shape (N x :attr:`output_dims`).
+        Parameters
+        ----------
+        values : array
+            The input values as numpy array of length :attr:`input_dims` or
```
Comment (QuLogic) [inline lib/matplotlib/transforms.py]:
```suggestion
            The input values as NumPy array of length :attr:`input_dims` or
```

File: lib/matplotlib/transforms.py [hunk 10]
```diff
@@ -1422,11 +1428,17 @@ def transform_non_affine(self, values):
...
+            shape (N x :attr:`input_dims`).
 
-        Alternatively, accepts a numpy array of length :attr:`input_dims`
-        and returns a numpy array of length :attr:`output_dims`.
+        Returns
+        -------
+        values : array
+            The output values as numpy array of length :attr:`input_dims` or
```
Comment (QuLogic) [inline lib/matplotlib/transforms.py]:
```suggestion
            The output values as NumPy array of length :attr:`input_dims` or
```

File: lib/matplotlib/scale.py [hunk 11]
```diff
@@ -354,18 +356,16 @@ class LogScale(ScaleBase):
 
     def __init__(self, axis, **kwargs):
         """
-        *basex*/*basey*:
+        Parameters
+        ----------
```
Comment (anntzer) [inline lib/matplotlib/scale.py]: indents of descriptions are not consistent

**Comment by timhoffm** [comment]
Not bothering to backport as 3.2 is approaching.

**Pull request merged**